src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -77,6 +77,7 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     if (!config.canAttemptToEstablishConnection()) {
```

Comment:
Suggestion: Can we consolidate the logic better in this method? There is also a check further down for `!config.canAttemptToEstablishConnection()` which will never be true.

Reply:
+1

please also add a unit test

Reply:
unnecessary checks removed for all plugins and unit tests added for this.